### PR TITLE
rework last-login/user-deletion to not be based on sql anymore

### LINKED ISSUE #352: Use Last Login plugin to determine last login time
https://doc.dovecot.org/configuration_manual/lastlogin_plugin/ can be configured to use doveauth as a dictionary via dictionary protocol.

Currently we update last login timestamp when passdb is requested, but this can happen even if the password is wrong. Last login plugin will make another doveauth request when successful login actually happens.

### PR DESCRIPTION
**Pull request by hpk42**
This pr moves all sqllite user-table state into a user's maildir's "password" file whose mtime reflects the last-login time. 

The PR also introduces a new "lastlogin" dictproxy used by the freshly enabled dovecot's lastlogin plugin to 

- to avoid updating the database on every login (it's a write operation, requires locks etc.) 
- to prevent outsiders to keep an e-mail address alive forever by periodically logging in with a wrong password (this today updates the last-login column, so the account never dies) 

Closes #352

### CODE REVIEW

File: chatmaild/src/chatmaild/delete_inactive_users.py [hunk 1]
```diff
@@ -2,28 +2,65 @@
...
+        os.utime(target, (timestamp, timestamp))
+    else:
+        if st.st_mtime < timestamp:
+            os.utime(target, (timestamp, timestamp))
+
+
+def get_last_login_from_userdir(userdir):
+    target = userdir.joinpath(LAST_LOGIN)
+    try:
+        return int(target.stat().st_mtime)
```
[the remark this answers is not in the document]
Comment (hpk42) [reply chatmaild/src/chatmaild/delete_inactive_users.py:39]: all time stamps are internally just ints, so i don't want to hand out a float here. 
get_day_timestamp also returns an int.

File: chatmaild/src/chatmaild/delete_inactive_users.py [hunk 2]
```diff
@@ -2,28 +2,71 @@
...
+        pending.append(userdir.name)
+        if len(pending) > chunksize:
+            clear_pending()
 
+    def clear_pending():
         with db.write_transaction() as conn:
-            for user in sublist:
+            for user in pending:
                 conn.execute("DELETE FROM users WHERE addr = ?", (user,))
+        pending[:] = []
```
[the remark this answers is not in the document]
Comment (hpk42) [reply chatmaild/src/chatmaild/delete_inactive_users.py:63]: oh, didn't notice this was added to the list type, thanks.

File: chatmaild/src/chatmaild/delete_inactive_users.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hpk42) [reply chatmaild/src/chatmaild/delete_inactive_users.py:39]: `st_mtime` seems to be a float. 

```
>>> import pathlib
>>> pathlib.Path("/etc/passwd").stat().st_mtime
1721045957.7741497
```

File: chatmaild/src/chatmaild/metadata.py [hunk 3]
```diff
@@ -117,9 +118,15 @@ def handle_dovecot_request(msg, transactions, notifier, metadata, iroh_relay=Non
             metadata.add_token_to_addr(addr, value)
         elif keyname[0] == "priv" and keyname[2] == "messagenew":
             notifier.new_message_for_addr(addr, metadata)
+        elif keyname[0] == "shared" and keyname[1] == "last-login":
```
[the remark this answers is not in the document]
Comment (hpk42) [reply chatmaild/src/chatmaild/metadata.py]: There now is a new lastlogin dictproxy, see edited top-level PR description.

**Review by missytake — APPROVED** [review]
woop woop DB go poof

**Pull request merged**